serverless.yml

```diff
@@ -1,228 +1,233 @@
-  - serverless-webpack
-  #  - serverless-offline-sqs
-  - serverless-offline
-  #  - serverless-offline-sns
-  #  - serverless-s3-local
-  - serverless-prune-plugin
+service: template-sls
+
+provider:
+  name: aws
+  runtime: nodejs12.x
+  environment:
+    STAGE: ${opt:stage, self:provider.stage}
+    ICONIK_TOKENS_TABLE_NAME: ${self:custom.tablesNames.IconikTokensTable.${opt:stage, self:provider.stage}}
+    USERS_TABLE_NAME: ${self:custom.tablesNames.UsersTable.${opt:stage, self:provider.stage}}
+    JOBS_TABLE_NAME: ${self:custom.tablesNames.JobsTable.${opt:stage, self:provider.stage}}
+  region: ${file(env.yml):${opt:stage, self:provider.stage}.REGION}
+  profile: ${file(env.yml):${opt:stage, self:provider.stage}.PROFILE}
+  tags:
+    client: ${file(env.yml):${opt:stage, self:provider.stage}.CLIENT}
+  iamRoleStatements:
+    - Effect: Allow
+      Action:
+        - cloudformation:DescribeStacks
+      Resource:
+        - arn:aws:cloudformation:*:*:stack/${file(env.yml):${opt:stage, self:provider.stage}.CLOUDFORMATION_STACK_NAME}/*
+    - Effect: Allow
+      Action:
+        - dynamodb:*
+      Resource:
+        - !GetAtt IconikTokensTable.Arn
+        - !Sub
+          - '${arn}/index/*'
+          - arn: !GetAtt IconikTokensTable.Arn
+    #        - !GetAtt UsersTable.Arn
+    #        - !Sub
+    #          - '${arn}/index/*'
+    #          - arn: !GetAtt UsersTable.Arn
+    #        - !GetAtt JobsTable.Arn
+    #        - !Sub
+    #          - '${arn}/index/*'
+    #          - arn: !GetAtt JobsTable.Arn
+    #    - Effect: Allow
+    #      Action:
+    #        - s3:*
+    #      Resource:
+    #        - arn:aws:s3:::${file(env.yml):${opt:stage, self:provider.stage}.BUCKET}
+    #        - arn:aws:s3:::${file(env.yml):${opt:stage, self:provider.stage}.BUCKET}/*
+    #    - Effect: Allow
+    #      Action:
+    #        - es:*
+    #      Resource:
+    #        - arn:aws:es:*:*:domain/${file(env.yml):${opt:stage, self:provider.stage}.ELASTIC_SEARCH_DOMAIN}
+    #        - arn:aws:es:*:*:domain/${file(env.yml):${opt:stage, self:provider.stage}.ELASTIC_SEARCH_DOMAIN}/*
+    #    - Effect: Allow
+    #      Action:
+    #        - ses:SendEmail
+    #      Resource: '*'
+    #    - Effect: Allow
+    #      Action:
+    #        - sqs:DeleteMessage
+    #        - sqs:ReceiveMessage
+    #        - sqs:SendMessage
+    #      Resource: arn:aws:sqs:*:*:${file(env.yml):${opt:stage, self:provider.stage}.SQS}
+    #    - Effect: Allow
+    #      Action:
+    #        - sns:CreatePlatformEndpoint
+    #        - sns:DeleteEndpoint
+    #        - sns:Publish
+    #      Resource:
+    #        - ${file(env.yml):${opt:stage, self:provider.stage}.SNS_ANDROID_APPLICATION_ARN}
+    #        - ${file(env.yml):${opt:stage, self:provider.stage}.SNS_IOS_APPLICATION_ARN}
+    - Effect: Allow
+      Action:
+        - events:PutRule
+        - events:DescribeRule
+        - events:PutTargets
+      Resource: '*'

```

Comment:
Could we use more specific resource pointer than '*'?

Reply:
No, because we create these resources after deployment with help lambda.

Reply:
Yes, you can be a bit more specific if you use something like this here
`arn:aws:events:*:*:rule/refresh-tokens-event`
`arn:aws:events:*:*:rule/invalidate-tokens-event`
And I recommend to add `refresh-tokens-event` and `invalidate-tokens-event` to custom variables and provide them in Lambda as env variables.